lib/trento/domain/sap_system/sap_system.ex

```diff
@@ -618,6 +702,18 @@ defmodule Trento.Domain.SapSystem do
        ),
        do: []
 
+  defp maybe_emit_sap_system_deregistered_event(
+         %SapSystem{
+           sap_system_id: sap_system_id,
+           database: %Database{
+             instances: nil
```

Comment:
Is this `instances: nil` or `instances: []` (or empty list)?
The same in other parts of the code.
Do we have any tests on this?